### PR TITLE
Geowave 811 rebase

### PR DESCRIPTION
**Pull request by mcarrier7**

### CODE REVIEW

File: core/store/src/main/java/mil/nga/giat/geowave/core/store/operations/config/AddStoreCommand.java [hunk 1]
```diff
@@ -114,11 +111,46 @@ public void execute(
...
+				for (Annotation annotation : field.getAnnotations()) {
+					if (annotation.annotationType() == Parameter.class) {
+						Parameter parameter = (Parameter) annotation;
+						if (JCommanderParameterUtils.isPassword(parameter)) {
+							String storeFieldName = getNamespace() + ".opts." + field.getName();
+							if (existingProps.containsKey(storeFieldName)) {
+								String value = existingProps.getProperty(storeFieldName);
+								String encryptedValue = value;
+								try {
+									encryptedValue = new SecurityUtils().encryptAndHexEncodeValue(value);
```
Comment (rfecher) [inline core/store/src/main/java/mil/nga/giat/geowave/core/store/operations/config/AddStoreCommand.java]: static method and pass in resourceLocation

File: examples/src/main/java/mil/nga/giat/geowave/examples/accumulo/app/GeoWaveDemoApp.java [hunk 2]
```diff
@@ -36,7 +37,10 @@ public static void main(
 		final boolean interactive = (System.getProperty("interactive") != null) ? Boolean.parseBoolean(System
 				.getProperty("interactive")) : true;
 
-		final String password = (System.getProperty("password") != null) ? System.getProperty("password") : "password";
+		// GeoWave:811 - providing ability to support encrypted passwords
+		final String password = new SecurityUtils()
```
Comment (rfecher) [inline examples/src/main/java/mil/nga/giat/geowave/examples/accumulo/app/GeoWaveDemoApp.java]: static method and pass in resourceLocation

File: extensions/cli/geoserver/src/main/java/mil/nga/giat/geowave/cli/geoserver/GeoServerConfig.java [hunk 3]
```diff
@@ -41,43 +40,58 @@ public GeoServerConfig(
...
-			gsConfig.setProperty(
+			gsConfigProperties.setProperty(
 					GEOSERVER_PASS,
 					pass);
 			update = true;
 		}
+		else {
+			try {
+				// if password in config props is encrypted, need to decrypt it
+				pass = new SecurityUtils().decryptHexEncodedValue(pass);
```
Comment (rfecher) [inline extensions/cli/geoserver/src/main/java/mil/nga/giat/geowave/cli/geoserver/GeoServerConfig.java]: static method and pass in resourceLocation

File: extensions/datastores/accumulo/src/main/java/mil/nga/giat/geowave/datastore/accumulo/BasicAccumuloOperations.java [hunk 4]
```diff
@@ -106,12 +107,22 @@ public BasicAccumuloOperations(
 		this(
 				null,
 				tableNamespace);
-		this.password = password;
+		// GeoWave:811 - providing ability to support encrypted passwords
+		try {
+			this.password = new SecurityUtils().decryptHexEncodedValue(password);
```
Comment (rfecher) [inline extensions/datastores/accumulo/src/main/java/mil/nga/giat/geowave/datastore/accumulo/BasicAccumuloOperations.java]: static method and pass in resourceLocation

File: extensions/datastores/accumulo/src/main/java/mil/nga/giat/geowave/datastore/accumulo/app/GeoWaveDemoAppShell.java [hunk 5]
```diff
@@ -27,7 +29,12 @@ public static void main(
 
 		final String instanceName = (System.getProperty("instanceName") != null) ? System.getProperty("instanceName")
 				: "geowave";
-		final String password = (System.getProperty("password") != null) ? System.getProperty("password") : "password";
+		// final String password = (System.getProperty("password") != null) ?
+		// System.getProperty("password") : "password";
+		// GeoWave:811 - providing ability to support encrypted passwords
+		final String password = new SecurityUtils()
```
Comment (rfecher) [inline extensions/datastores/accumulo/src/main/java/mil/nga/giat/geowave/datastore/accumulo/app/GeoWaveDemoAppShell.java]: static method and pass in resourceLocation

File: extensions/datastores/accumulo/src/main/java/mil/nga/giat/geowave/datastore/accumulo/util/ConnectorPool.java [hunk 6]
```diff
@@ -29,19 +33,30 @@ public synchronized Connector getConnector(
 			final String password )
 			throws AccumuloException,
 			AccumuloSecurityException {
+
+		String decryptedPassword = password;
+		try {
+			decryptedPassword = new SecurityUtils().decryptHexEncodedValue(password);
```
Comment (rfecher) [inline extensions/datastores/accumulo/src/main/java/mil/nga/giat/geowave/datastore/accumulo/util/ConnectorPool.java]: static method and pass in resourceLocation

File: services/webapp/src/main/java/mil/nga/giat/geowave/service/impl/GeoserverServiceImpl.java [hunk 7]
```diff
@@ -85,6 +87,15 @@ public GeoserverServiceImpl(
 				props,
 				"geoserver.password");
 
+		try {
+			geoserverPass = new SecurityUtils().decryptHexEncodedValue(geoserverPass);
```
Comment (rfecher) [inline services/webapp/src/main/java/mil/nga/giat/geowave/service/impl/GeoserverServiceImpl.java]: static method and pass in resourceLocation

File: core/cli/src/main/java/mil/nga/giat/geowave/core/cli/converters/GeoWaveBaseConverter.java [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+		String password = new String(
+				passwordChars);
+		return decryptValue(password);
+	}
+
+	protected static String encryptValue(
+			String value ) {
+		if (value != null) {
+			try {
+				return new SecurityUtils().encryptAndHexEncodeValue(value);
```
Comment (rfecher) [inline core/cli/src/main/java/mil/nga/giat/geowave/core/cli/converters/GeoWaveBaseConverter.java]: static method and pass in resourceLocation

File: core/cli/src/main/java/mil/nga/giat/geowave/core/cli/converters/GeoWaveBaseConverter.java [hunk 9]
```diff
@@ -0,0 +1,237 @@
...
+			}
+		}
+		return value;
+	}
+
+	protected static String decryptValue(
+			String value ) {
+		if (value != null) {
+			try {
+				return new SecurityUtils().decryptHexEncodedValue(value);
```
Comment (rfecher) [inline core/cli/src/main/java/mil/nga/giat/geowave/core/cli/converters/GeoWaveBaseConverter.java]: static method and pass in resourceLocation

File: core/cli/src/main/java/mil/nga/giat/geowave/core/cli/operations/config/security/NewTokenCommand.java [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+							Set<Object> keySet = configProps.keySet();
+							Iterator<Object> keyIter = keySet.iterator();
+							if (keyIter != null) {
+								String configKey = null;
+								while (keyIter.hasNext()) {
+									configKey = (String) keyIter.next();
+									String configValue = configProps.getProperty(configKey);
+									if (configValue != null && !"".equals(configValue.trim())
+											&& BaseEncryption.isProperlyWrapped(configValue)) {
+										String decryptedValue = new SecurityUtils().decryptHexEncodedValue(
```
Comment (rfecher) [inline core/cli/src/main/java/mil/nga/giat/geowave/core/cli/operations/config/security/NewTokenCommand.java]: static method and pass in resourceLocation

File: core/cli/src/main/java/mil/nga/giat/geowave/core/cli/operations/config/security/NewTokenCommand.java [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+												configValue,
+												backupFile.getCanonicalPath());
+										String encryptedValue = new SecurityUtils().encryptAndHexEncodeValue(
```
Comment (rfecher) [inline core/cli/src/main/java/mil/nga/giat/geowave/core/cli/operations/config/security/NewTokenCommand.java]: static method and pass in resourceLocation

File: core/cli/src/test/java/mil/nga/giat/geowave/core/cli/operations/config/security/SecurityUtilsTest.java [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+ */
+public class SecurityUtilsTest
+{
+	@Test
+	public void testEncryptionDecryption()
+			throws Exception {
+		String rawInput = "geowave";
+
+		File tokenFile = new File(
+				new SecurityUtils().getResourceLocation());
```
Comment (rfecher) [inline core/cli/src/test/java/mil/nga/giat/geowave/core/cli/operations/config/security/SecurityUtilsTest.java]: find a different way to get default resource location

File: core/store/src/main/java/mil/nga/giat/geowave/core/store/operations/config/AddStoreCommand.java [hunk 13]
```diff
@@ -114,11 +111,46 @@ public void execute(
...
+			pluginOptions.getFactoryOptions().validatePluginOptions(
+					existingProps);
+		}
+
 		// Save the store options.
 		pluginOptions.save(
 				existingProps,
 				getNamespace());
 
+		if (pluginOptions.getFactoryOptions() != null) {
```
Comment (rfecher) [inline core/store/src/main/java/mil/nga/giat/geowave/core/store/operations/config/AddStoreCommand.java]: add this logic to write properties file method (and decrypt immediately on read).  those should be the only 2 places where we need to worry about encrypt/decrypt (as far as I know, I could stand corrected).

File: extensions/datastores/accumulo/src/main/java/mil/nga/giat/geowave/datastore/accumulo/operations/config/AccumuloPasswordConverter.java [hunk 14]
```diff
@@ -0,0 +1,23 @@
+/**
```
Comment (rfecher) [inline extensions/datastores/accumulo/src/main/java/mil/nga/giat/geowave/datastore/accumulo/operations/config/AccumuloPasswordConverter.java]: this class seems unnecessary (and is not currently used).  I think we can safely get rid of it.

File: extensions/cli/geoserver/src/main/java/mil/nga/giat/geowave/cli/geoserver/GeoServerPasswordConverter.java [hunk 15]
```diff
@@ -0,0 +1,21 @@
+/**
```
Comment (rfecher) [inline extensions/cli/geoserver/src/main/java/mil/nga/giat/geowave/cli/geoserver/GeoServerPasswordConverter.java]: this class seems unnecessary (and is not currently used).  I think we can safely get rid of it.

File: extensions/datastores/accumulo/src/main/java/mil/nga/giat/geowave/datastore/accumulo/operations/config/AccumuloRequiredOptions.java [hunk 16]
```diff
@@ -25,25 +27,27 @@
 	@Parameter(names = {
 		"-z",
 		"--" + ZOOKEEPER_CONFIG_KEY
-	}, description = "A comma-separated list of zookeeper servers that an Accumulo instance is using", required = true)
+	}, description = "A comma-separated list of zookeeper servers that an Accumulo instance is using", converter = RequiredFieldConverter.class)
```
Comment (rfecher) [inline extensions/datastores/accumulo/src/main/java/mil/nga/giat/geowave/datastore/accumulo/operations/config/AccumuloRequiredOptions.java]: seems like one of the few places we use this.  I understand it gives different functionality (prompting a user).  Do we want this functionality to be consistently used throughout?  Just seems like we should either do this for other current required options or not, or have some logic for when to use this vs. the jcommander annotation.

File: extensions/datastores/accumulo/src/test/java/mil/nga/giat/geowave/datastore/accumulo/operations/config/AccumuloRequiredOptionsTest.java [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ */
+public class AccumuloRequiredOptionsTest
+{
+	/**
+	 * @param args
+	 */
+	public static void main(
+			String[] args ) {
+		AccumuloRequiredOptions test = new AccumuloRequiredOptions();
```
Comment (rfecher) [inline extensions/datastores/accumulo/src/test/java/mil/nga/giat/geowave/datastore/accumulo/operations/config/AccumuloRequiredOptionsTest.java]: can we just delete this class?

Comment (rfecher) [inline extensions/datastores/accumulo/src/test/java/mil/nga/giat/geowave/datastore/accumulo/operations/config/AccumuloRequiredOptionsTest.java]: or we can turn it into a unit test if preferred?

File: core/cli/src/main/java/mil/nga/giat/geowave/core/cli/converters/PasswordConverter.java [hunk 18]
```diff
@@ -0,0 +1,241 @@
...
+				// update the value in the configs properties file
+				if (updatePasswordInConfigs() && getPropertyKey() != null && !"".equals(getPropertyKey().trim())) {
+					Properties configProps = getGeoWaveConfigProperties();
+					if (configProps != null) {
+						// encrypt the value to be stored in the properties
+						convertedValue = encryptValue(convertedValue);
+						configProps.put(
+								getPropertyKey(),
+								convertedValue);
+						ConfigOptions.writeProperties(
```
Comment (rfecher) [inline core/cli/src/main/java/mil/nga/giat/geowave/core/cli/converters/PasswordConverter.java]: seems outside of the scope of a "converter" to write to the properties file here.

File: core/cli/src/main/java/mil/nga/giat/geowave/core/cli/operations/config/options/ConfigOptions.java [hunk 19]
```diff
@@ -50,19 +49,19 @@
 		"-cf",
 		"--config-file"
 	}, description = "Override configuration file (default is <home>/.geowave/config.properties)")
-	private String configFile;
+	private static String configFile;
```
Comment (rfecher) [inline core/cli/src/main/java/mil/nga/giat/geowave/core/cli/operations/config/options/ConfigOptions.java]: we should move away from using a static reference here...in practice in may not be an issue, but for example if there ever were different instances of ConfigOptions that used different values for configFile, this static member variable would be a major problem.

File: core/cli/src/main/java/mil/nga/giat/geowave/core/cli/operations/config/security/NewTokenCommand.java [hunk 20]
```diff
@@ -0,0 +1,110 @@
...
+												encryptedValue);
+										updated = true;
+									}
+								}
+							}
+							if (updated) {
+								ConfigOptions.writeProperties(
+										getGeoWaveConfigFile(params),
+										configProps);
+								System.out.println("GeoServer Config Saved");
```
Comment (rfecher) [inline core/cli/src/main/java/mil/nga/giat/geowave/core/cli/operations/config/security/NewTokenCommand.java]: we can log if necessary, is there a reason for system.out here?

File: core/cli/src/test/java/mil/nga/giat/geowave/core/cli/operations/config/security/SecurityUtilsTest.java [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+		if (tokenFile != null && tokenFile.exists()) {
+			String encryptedValue = new SecurityUtils().encryptAndHexEncodeValue(
+					rawInput,
+					tokenFile.getCanonicalPath());
+			System.out.println("encryptedValue: " + encryptedValue);
```
Comment (rfecher) [inline core/cli/src/test/java/mil/nga/giat/geowave/core/cli/operations/config/security/SecurityUtilsTest.java]: we can log if necessary, is there a reason for system.out here?

File: core/cli/src/test/java/mil/nga/giat/geowave/core/cli/operations/config/security/SecurityUtilsTest.java [hunk 22]
```diff
@@ -0,0 +1,41 @@
...
+
+			String decryptedValue = new SecurityUtils().decryptHexEncodedValue(
+					encryptedValue,
+					tokenFile.getCanonicalPath());
+			System.out.println("decryptedValue: " + decryptedValue);
```
Comment (rfecher) [inline core/cli/src/test/java/mil/nga/giat/geowave/core/cli/operations/config/security/SecurityUtilsTest.java]: we can log if necessary, is there a reason for system.out here?

File: extensions/cli/geoserver/src/main/java/mil/nga/giat/geowave/cli/geoserver/GeoServerConfig.java [hunk 23]
```diff
@@ -86,7 +100,7 @@ public GeoServerConfig(
 		if (update) {
 			ConfigOptions.writeProperties(
 					propFile,
-					gsConfig);
+					gsConfigProperties);
 
 			System.out.println("GeoServer Config Saved");
```
Comment (rfecher) [inline extensions/cli/geoserver/src/main/java/mil/nga/giat/geowave/cli/geoserver/GeoServerConfig.java]: we can log if necessary, is there a reason for system.out here?

File: extensions/cli/geoserver/src/main/java/mil/nga/giat/geowave/cli/geoserver/GeoServerConfig.java [hunk 24]
```diff
@@ -96,11 +110,10 @@ public GeoServerConfig(
...
-		this.pass = DEFAULT_PASS;
-		this.url = DEFAULT_URL;
-		this.workspace = DEFAULT_WORKSPACE;
+		this(
+				new File(
+						System.getProperty("user.home") + File.separator + ConfigOptions.GEOWAVE_CACHE_PATH
+								+ File.separator + ConfigOptions.GEOWAVE_CACHE_FILE));
 	}
 
 	public String getUrl() {
```
Comment (rfecher) [inline extensions/cli/geoserver/src/main/java/mil/nga/giat/geowave/cli/geoserver/GeoServerConfig.java]: now that we are checking for https within the client, it seems like it may make sense to use `https://` on line 124 if we detect a port that ends with `443`

File: extensions/cli/geoserver/src/main/java/mil/nga/giat/geowave/cli/geoserver/GeoServerConfig.java (hunk 23 above)
Comment (mcarrier7) [inline extensions/cli/geoserver/src/main/java/mil/nga/giat/geowave/cli/geoserver/GeoServerConfig.java]: I can switch to a log statement. The system.out was there previously and I didn't think to change it
